src/osdc/Journaler.cc

```diff
@@ -1606,3 +1614,18 @@ void Journaler::shutdown()
   waitfor_safe.clear();
 }
 
+int Journaler::check_isreadable()
+{
+  int r=0;
+  std::unique_lock l(replaylock,std::defer_lock);
+  l.lock();
+  while (!is_readable() &&
+      get_read_pos() < get_write_pos() &&
+      !get_error()) {
+    C_SaferCond_ReferenceLock readable_waiter(replaylock);
```

Comment:
Would it not be sufficient to use `C_SaferCond`?

Reply:
It can work well too, and the modification is smaller.

![image](https://user-images.githubusercontent.com/44566632/196089917-53487b4d-fcb5-485d-9b09-21c3f9acb7d4.png)

Maybe I should alter the commit, should I ?

Reply:
I don't see an issue with using `C_SaferCond` - let's switch using that. Also, any reason to use `std::defer_lock`?